### DISCUSSION TITLE
Is it possible to extract individual unconnected line segments from PyVista mesh of lines

### DISCUSSION THREAD

**Discussion by 1939938853** [pyvista/pyvista#5275] [Q&A] [answered]
Hello

Just wonder if it is  possible to extract individual unconnected line segments from PyVista mesh of lines? How?

I want to extract the individual holes from the output of  PyMeshFix [MeshFix.extract_holes()](https://pymeshfix.pyvista.org/examples/cow.html#sphx-glr-examples-cow-py). The output is PyVista mesh of lines. Or to extract the individual featured edge group. for example [here](https://docs.pyvista.org/version/stable/examples/01-filter/extract-edges.html#extract-edges-example). 

Thank you!

![sphx_glr_cow_001](https://github.com/pyvista/pyvista/assets/59957922/0285bb9c-337c-4e75-961d-26a2f443043b)

![image](https://github.com/pyvista/pyvista/assets/59957922/44dbe203-427c-4269-bcaa-31232b9297f0)

**Comment by edabor** [comment, marked as answer]
Hi !

I guess the connectivity filter is what you need ? You can then threshold on the `RegionId` cell-data to get the hole loops you want.

https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSetFilters.connectivity.html#pyvista.DataSetFilters.connectivity
https://docs.pyvista.org/version/stable/examples/01-filter/connectivity.html#connectivity

```python
meshfix = mf.MeshFix(holy_cow)
holes = meshfix.extract_holes().connectivity()
```

**Comment by 1939938853** [comment]
Thanks. I saw that connectivity() will create an array with a name of 'RegionId'. This can be a label array used  to group cells. Am I right?

next step is how to extract cell and points with this RegionId array
